test/python/test_qobj.py

```diff
@@ -60,38 +55,31 @@ def test_as_dict_against_schema(self):
         with open(file_path, 'r') as schema_file:
             schema = json.load(schema_file)
 
-        passed = False
         try:
-            jsonschema.validate(self._valid_qobj.as_dict(), schema)
-            passed = True
-        except jsonschema.ValidationError as err:
-            self.log.info('Error validating Qobj to_dict with JSON schema: %s', err)
-
-        self.assertTrue(passed)
-
-
-class TestQobjConfig(QiskitTestCase):
-    """Tests for QobjConfig."""
-    def test_init_qobj_config(self):
-        """Test initialization of a QobjConfig for required arguments."""
-        with self.assertRaises(TypeError):
-            # pylint: disable=no-value-for-parameter
-            QobjConfig()
-
-
-class TestQobjExperiment(QiskitTestCase):
-    """Tests for QobjExperiment."""
-    def test_init_qobj_experiment(self):
-        """Test initialization of a QobjExperiment for required arguments."""
-        with self.assertRaises(TypeError):
-            # pylint: disable=no-value-for-parameter
-            QobjExperiment()
+            jsonschema.validate(self.valid_qobj.as_dict(), schema)
+        except jsonschema.ValidationError as validation_error:
+            self.fail(str(validation_error))
+
+    @unittest.expectedFailure  # expected to fail until _qobjectify_item is updated
+    def test_from_dict_per_class(self):
+        """Test Qobj and its subclass representations given a dictionary."""
+        qobj_config_dict = {'shots': 1, 'memory_slots': 2}
+        qobj_experiment_dict = {'instructions':
+                                [QobjInstruction(name='u1', qubits=[1], params=[0.4])]}
+        qobj_instruction_dict = {'name': 'u1', 'qubits': [1], 'params': [0.4]}
+
+        valid_qobj_config = QobjConfig(shots=1024, memory_slots=2, max_credits=10)
+        valid_qobj_experiment = QobjExperiment(
+            instructions=[QobjInstruction(name='u1', qubits=[1], params=[0.4])]
+        )
+        valid_qobj_instruction = QobjInstruction(name='u1', qubits=[1], params=[0.4])
 
+        dicts = [self.valid_dict, qobj_config_dict, qobj_experiment_dict, qobj_instruction_dict]
+        objects = [self.valid_qobj, valid_qobj_config,
+                   valid_qobj_experiment, valid_qobj_instruction]
+        classes = [Qobj, QobjConfig, QobjExperiment, QobjInstruction]
```

Comment:
Non-blocking but a recommendation: if you create a more explicit data structure, you won’t need to zip it after, you increase the readability of the test (avoiding the overhead introduced by many variables), and you won’t rely on indices to keep related parameters together:

```python
test_parameters = {
  QobjConfig: (
    QobjConfig(shots=1024, memory_slots=2, max_credits=10),
    {'shots': 1, 'memory_slots': 2}
  ),
  ...
}
```

Then you can iterate it by doing:

```python
for class_, (qobj, expected_dict) in test_parameters.items():
  with self.subTest(...):
    # test things
```